galley/pkg/config/analysis/analyzers/auth/mtls/destination_rule_checker.go

```diff
@@ -0,0 +1,215 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mtls
+
+import (
+	"sort"
+
+	"istio.io/istio/galley/pkg/config/resource"
+
+	"istio.io/api/networking/v1alpha3"
+	"istio.io/istio/galley/pkg/config/analysis/analyzers/util"
+	"istio.io/istio/pkg/config/host"
+)
+
+// DestinationRuleChecker computes whether or not MTLS is used according to
+// DestinationRules added to the instance. It handles the complicated logic of
+// looking up which DestinationRule takes effect for a given source namespace,
+// destination namespace and host name.
+type DestinationRuleChecker struct {
```

Comment:
Maybe add a link or reference to the logic in Pilot that this mimics? Or to the doc that defines all this? (In other words, what's the source of truth for the logic that we're using?)